scripting/messages_test.sp

```diff
@@ -0,0 +1,63 @@
+#include <spmod>
+
+public PluginInfo pluginInfo =
+{
+	name = "Message test",
+	version = "0.0.0",
+	author = "author",
+	url = "https://github.com/Amaroq7/SPMod"
+};
+
+MessageHook h1, h2;
+
+public void OnPluginInit()
+{
+	Command("say /mblock", MsgTest);
+	Command("say /h1", MsgTesth1);
+	Command("say /h11", MsgTesth11);
+	h1 = HookMessage(GetUserMsgId("SayText"), MessageHandler, false);
```

Comment:
General comment: I don't like the complexity we introduce with caching message ids in global variables (well, not complexity, but boilerplate). Probably we can create `enum` with message names? (with mapping to real ids inside natives of course) This also prevents users from typos, like `"CurWaepon"`, so it will be caught by the compiler at compile-time, not at run-time. Not sure if we need to have both types of message ids (original ones + `enum`). Having both will help in case of exotic mods for which user can just specify message name instead of proposing new one to the `enum`.

Reply:
I like the idea, we could both as you mentioned user may use some exotic mod and I prefer to have future proof code.